### PR TITLE
Use latest installer for upgrade tests in installer

### PR DESCRIPTION
**Pull request by vrutkovs**
`openshift-*-upgrade` workflow installs the cluster using installer from `stable-initial:installer` and updates it to `release:latest`. Installer PR presubmits tag build proposed change in `stable:installer` image, so it doesn't get used there.

This PR adds new workflows - `openshift-aws/gcp-upgrade-latestinstaller` - which use `stable:installer` (by using `- chain: ipi-aws-pre` instead of default `- chain: ipi-aws-pre-stableinitial`)and run upgrade test. 

Its not meant to be used anywhere outside of installer PR tests.

### CODE REVIEW

File: ci-operator/step-registry/openshift/upgrade/gcp/latestinstaller/openshift-upgrade-gcp-latestinstaller-workflow.yaml [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+    allow_best_effort_post_steps: true
+    pre:
+    - ref: ipi-install-hosted-loki
+    - chain: ipi-gcp-pre
+    test:
+    - ref: openshift-e2e-test
+    post:
+    - chain: ipi-gcp-post
+    dependencies:
+      OPENSHIFT_INSTALL_RELEASE_IMAGE_OVERRIDE: "release:initial"
```
Comment (smarterclayton) [inline ci-operator/step-registry/openshift/upgrade/gcp/latestinstaller/openshift-upgrade-gcp-latestinstaller-workflow.yaml:13]: What actually changed from the job?

Comment (vrutkovs) [reply ci-operator/step-registry/openshift/upgrade/gcp/latestinstaller/openshift-upgrade-gcp-latestinstaller-workflow.yaml:13]: This workflow uses
`- chain: ipi-aws-pre` (same as clean install), while `openshift-e2e-upgrade` uses
`- chain: ipi-aws-pre-stableinitial`

Updated description too

**Review by ravisantoshgudimetla — APPROVED** [review]
LGTM

Thank you for working on this @vrutkovs

File: ci-operator/step-registry/openshift/upgrade/aws/latestinstaller/openshift-upgrade-aws-latestinstaller-workflow.yaml [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+    dependencies:
+      OPENSHIFT_INSTALL_RELEASE_IMAGE_OVERRIDE: "release:initial"
+      OPENSHIFT_UPGRADE_RELEASE_IMAGE_OVERRIDE: "release:latest"
+    env:
+      TEST_TYPE: upgrade
+      TEST_SUITE: ""
+      TEST_UPGRADE_OPTIONS: ""
+      DELETE_MC: "false"
+  documentation: |-
+    The Openshift E2E AWS workflow executes the upgrade end-to-end test suite on AWS with a default cluster configuration and Loki as log collector, exercising an upgrade during the process.
```
Comment (ravisantoshgudimetla) [inline ci-operator/step-registry/openshift/upgrade/aws/latestinstaller/openshift-upgrade-aws-latestinstaller-workflow.yaml:22]: Perhaps you can update this comment that this workflow has to be used within installer repo?

Comment (vrutkovs) [reply ci-operator/step-registry/openshift/upgrade/aws/latestinstaller/openshift-upgrade-aws-latestinstaller-workflow.yaml:22]: I added a note right below, not sure if its sufficient or need to be mentioned earlier

File: ci-operator/step-registry/openshift/upgrade/aws/latestinstaller/openshift-upgrade-aws-latestinstaller-workflow.yaml [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+    - ref: ipi-install-hosted-loki
+    - chain: ipi-aws-pre
+    test:
+    - ref: openshift-e2e-test
+    post:
+    - ref: openshift-e2e-test-capabilities-check
+    - chain: ipi-aws-post
+    dependencies:
+      OPENSHIFT_INSTALL_RELEASE_IMAGE_OVERRIDE: "release:initial"
+      OPENSHIFT_UPGRADE_RELEASE_IMAGE_OVERRIDE: "release:latest"
```
Comment (wking) [inline ci-operator/step-registry/openshift/upgrade/aws/latestinstaller/openshift-upgrade-aws-latestinstaller-workflow.yaml:15]: Do we need a new workflow for this, vs. overriding `dependencies` for the default workflow [like this][1]?

[1]: https://github.com/openshift/release/blob/f3cbb682c9174082641a1c2fb17d15045126377a/ci-operator/config/openshift/kubernetes/openshift-kubernetes-release-4.9.yaml#L259-L260

Comment (vrutkovs) [reply ci-operator/step-registry/openshift/upgrade/aws/latestinstaller/openshift-upgrade-aws-latestinstaller-workflow.yaml:15]: We need a new workflow to use different chain

Comment (vrutkovs) [reply ci-operator/step-registry/openshift/upgrade/aws/latestinstaller/openshift-upgrade-aws-latestinstaller-workflow.yaml:15]: Overriding deps would be confusing imo, but possible, yeah.

I'd prefer a separate workflow

Comment (vrutkovs) [reply ci-operator/step-registry/openshift/upgrade/aws/latestinstaller/openshift-upgrade-aws-latestinstaller-workflow.yaml:15]: Overriding `openshift-e2e-test` dependencies won't change imported imagestreams, so openshift-upgrade-aws workflow would still use `stable-initial:installer`

**Review by wking — APPROVED** [review]
I really wish we could tie step images to different release images per job, but since we can't:

/lgtm

File: ci-operator/step-registry/openshift/upgrade/aws/latestinstaller/openshift-upgrade-aws-latestinstaller-workflow.metadata.json [hunk 4]
```diff
@@ -0,0 +1,17 @@
+{
+	"path": "openshift/upgrade/aws/latestinstaller/openshift-upgrade-aws-latestinstaller-workflow.yaml",
+	"owners": {
+		"approvers": [
+			"smarterclayton",
+			"wking",
+			"stevekuznetsov",
```
Comment (wking) [inline ci-operator/step-registry/openshift/upgrade/aws/latestinstaller/openshift-upgrade-aws-latestinstaller-workflow.metadata.json:7]: I think we need to [drop some of these][1].

/hold

[1]: https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/openshift_release/17987/pull-ci-openshift-release-master-step-registry-metadata/1389782447410909184#1:build-log.txt%3A17

**Comment by smarterclayton** [comment]
Rehearse failures not related to PR?

**Comment by vrutkovs** [comment]
Lets give those another try

/retest

**Pull request merged**